### PR TITLE
[TTAHUB-5403] Update start and end date filters to support TR

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

This PR updates the start and end date filters on the TTA History tab of the RTR to also filter the TR's by their respective dates. It looks like we were already doing this for the start date so we just needed to add the end date.

## How to test

- Review the code
- Make sure the filter for both start and end date also filter the TR's.

## Jira Issue(s)

* https://jira.acf.gov/browse/TTAHUB-5403

### Every PR

- [x] Linked Jira issue
- [x] JIRA issue status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete
- [ ] QA review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] PR created as **Draft**
- [ ] Staging smoke test completed
- [ ] PR transitioned to **Open** _(this `ready_for_review` transition triggers the Slack/Jira automation)_
- [ ] Reviewer added after the PR is **Open** _(`elainaparrish` is the authorized approver under normal circumstances)_
  - _Sequence: Draft PR → Smoke test → Open PR (automation runs) → Add reviewer_
  - _Confirm that the Slack notification was sent after the PR was opened_
  - _Confirm that linked Jira ticket(s) transitioned as expected; if not, review the GitHub Actions workflow logs_

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: src/scopes/trainingReports/endDate.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  return parsed.format('YYYY-MM-DD');
+}
+
+export function beforeEndDate(date) {
+  const converted = toIsoFormat(date[0]);
+  if (!converted) return {};
+  // Use TO_DATE to convert stored MM/DD/YYYY to a proper date for comparison
+  return {
+    [Op.and]: [
+      sequelize.literal(
```
Comment (kryswisnaskas) [inline src/scopes/trainingReports/endDate.js]: Codex said: High: `src/scopes/trainingReports/endDate.js:33`, `:46`, `:66` parses stored `EventReportPilot.data.endDate` with `TO_DATE(..., 'MM/DD/YYYY')` unguarded. A single TR event with ISO, short-year, or malformed JSONB date can make any end-date filtered TTA History/widget query fail. Fix with a shared safe date expression, like the guarded CASE pattern in session report sorting, returning `NULL` for unparseable values, and add tests for `YYYY-MM-DD`, `M/D/YY`, empty, and invalid stored dates.

File: src/scopes/trainingReports/endDate.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        `TO_DATE(NULLIF("EventReportPilot"."data"->>'endDate', ''), 'MM/DD/YYYY') >= ${sequelize.escape(converted)}::date`
+      ),
+    ],
+  };
+}
```
Comment (kryswisnaskas) [inline src/scopes/trainingReports/endDate.js:9]: Codex said: Medium: `src/scopes/trainingReports/endDate.js:51` splits ranges with `dates[0].split('-')`, but the same file advertises `YYYY-MM-DD` and `YYYY-M-D` as accepted input formats. `endDate.win=2026-03-01-2026-03-31` silently becomes `{}` and drops the filter, broadening results. Either parse ranges with an unambiguous contract or reject unsupported dashed ranges explicitly; add a regression test for dashed range input.

File: src/scopes/trainingReports/index.js [hunk 3]
```diff
@@ -37,6 +38,12 @@ export const topicToQuery = {
     in: (query) => withStandard(query),
```
Comment (kryswisnaskas) [inline src/scopes/trainingReports/index.js:38]: Codex said: Open Question: The new scope filters `EventReportPilot.data.endDate`, while widgets count included `SessionReportPilot` rows. If the TTA History date filters are intended to filter session dates, the session include needs its own `data.endDate` predicate too.

File: src/scopes/trainingReports/endDate.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+export function beforeEndDate(date) {
+  const converted = toIsoFormat(date[0]);
+  if (!converted) return {};
+  // Use TO_DATE to convert stored MM/DD/YYYY to a proper date for comparison
+  return {
+    [Op.and]: [
+      sequelize.literal(
+        `TO_DATE(NULLIF("EventReportPilot"."data"->>'endDate', ''), 'MM/DD/YYYY') <= ${sequelize.escape(converted)}::date`
```
Comment (kryswisnaskas) [inline src/scopes/trainingReports/endDate.js]: **Claude said:**

**(Medium) The actual robustness fix — `NULLIF(..., '')` — is untested**

The most consequential change is adding `NULLIF("…"->>'endDate', '')` (and retrofitting it onto `startDate.js`). Its entire purpose is to stop empty-string dates from reaching `TO_DATE`. Yet **no test creates a report with an empty, null, or missing `endDate`** — every fixture has a valid `MM/DD/YYYY` value. Events very commonly have a blank `endDate` (not yet entered), so this is the realistic case the guard exists for, and it's the one with zero coverage.

Add a fixture and assertions, e.g.:

```js
emptyEndDateEventReportPilot = await EventReportPilot.create({
  ownerId: mockUser.id, pocIds: [mockUser.id], collaboratorIds: [],
  regionId: mockUser.homeRegionId,
  data: { endDate: '' }, // also worth a variant with endDate omitted entirely
});
```

Then assert it is **excluded** from `bef`/`aft`/`win` results and, critically, that the query doesn't throw. Without this, a regression that drops the `NULLIF` would pass CI. The same gap now exists for `startDate.js`, which this PR also modified.

File: src/scopes/trainingReports/endDate.js
Comment (thewatermethod) [inline src/scopes/trainingReports/endDate.js:1]: This file is pretty close to a duplicate of `src/scopes/trainingReports/startDate.js` can we extract to some kind of date utils or common utils for maintainability?

**Review by thewatermethod — APPROVED** [review]

File: src/scopes/trainingReports/dateUtils.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  return `(
+    CASE
+      WHEN NULLIF(${path}, '') IS NULL THEN NULL
+      WHEN ${path} ~ '^\\d{4}-\\d{1,2}-\\d{1,2}$' THEN (${path})::date
+      WHEN ${path} ~ '^\\d{1,2}/\\d{1,2}/\\d{2}$' THEN TO_DATE(${path}, 'MM/DD/YY')
+      WHEN ${path} ~ '^\\d{1,2}/\\d{1,2}/\\d{4}$' THEN TO_DATE(${path}, 'MM/DD/YYYY')
+      ELSE NULL
+    END
+  )`;
+}
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply src/scopes/trainingReports/dateUtils.js:69]: I agree this still needs a guard. The current CASE checks only the string shape, so ISO-shaped but invalid stored values like `2021-99-99` or `2021-02-31` can still throw at `(${path})::date` and break the whole filtered query. The new tests cover malformed text, but not malformed calendar dates that match the regex.

Can we make the stored-date parser return `NULL` for semantically invalid dates too, and add regression coverage for invalid ISO-shaped dates for both `startDate` and `endDate`?

**Comment by thewatermethod** [comment]
I approved, looks good, but FWIW: there's also [this PR](https://github.com/HHS/Head-Start-TTADP/pull/3529) which may cut down on some of the back and forth on sanitizing for a JSON field -- fine to ignore if we're already too far down this path
